torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -278,19 +290,31 @@ class ProcessGroupNCCL : public ProcessGroup {
     // this callback. This new FutureNCCL's cudaEvents will record the
     // callback's stream and will have the result value of the callback.
     void addCallback(std::function<void(void)> callback) override {
-      // FIXME Should we find a way to allow to change the priority of streams?
-      at::cuda::CUDAStream stream =
-          at::cuda::getStreamFromPool(/*isHighPriority=*/false, deviceIndex_);
+      // Get a stream for all devices, even those that are not used by the
+      // value, because the user's callback could use those other devices.
+      std::vector<at::cuda::CUDAStream> streams;
+      for (c10::DeviceIndex idx = 0; idx < c10::cuda::device_count(); idx++) {
+        // FIXME Should we find a way to allow to change the priority of
```

Comment:
when do we need high-priority streams?

Reply:
I actually have no idea. Do you know when high- vs low-priority streams were used in ProcessGroupNCCL? What was the reason? Does it still apply here?

Reply:
I haven't never seen a user trying to configure stream priority for NCCL ops. And I don't think it's possible with the `init_process_group` API. Users probably will have to use the un-documented `ProcessGroupNCCL` ctor API. 

Besides, I am also not sure how much impact the stream priority can have on the schedule and how visible that is to the e2e perf. 

If there is no specific use case for now, we probably can keep it simple for now? It will be interesting to run some experiments to quantify its impact on perf in the future.